lib/iris/fileformats/netcdf.py

```diff
@@ -1825,166 +1825,57 @@ def _get_coord_variable_name(self, cube, coord):
-        data = dimensional_metadata.data
-
-        if isinstance(dimensional_metadata, iris.coords.CellMeasure):
-            # Disallow saving of *masked* cell measures.
-            # NOTE: currently, this is the only functional difference required
-            # between variable creation for an ancillary and a cell measure.
-            if ma.is_masked(data):
-                # We can't save masked points properly, as we don't maintain a
-                # suitable fill_value.  (Load will not record one, either).
-                msg = "Cell measures with missing data are not supported."
-                raise ValueError(msg)
-
-        # Get the values in a form which is valid for the file format.
-        data = self._ensure_valid_dtype(
-            data, "coordinate", dimensional_metadata
-        )
-
-        # Create the CF-netCDF variable.
-        cf_var = self._dataset.createVariable(
-            cf_name, data.dtype.newbyteorder("="), cf_dimensions
-        )
-
-        # Add the data to the CF-netCDF variable.
-        cf_var[:] = data
-
-        if dimensional_metadata.units.is_udunits():
-            _setncattr(cf_var, "units", str(dimensional_metadata.units))
-
-        if dimensional_metadata.standard_name is not None:
-            _setncattr(
-                cf_var, "standard_name", dimensional_metadata.standard_name
-            )
-
-        if dimensional_metadata.long_name is not None:
-            _setncattr(cf_var, "long_name", dimensional_metadata.long_name)
-
-        # Add any other custom coordinate attributes.
-        for name in sorted(dimensional_metadata.attributes):
-            value = dimensional_metadata.attributes[name]
-
-            # Don't clobber existing attributes.
-            if not hasattr(cf_var, name):
-                _setncattr(cf_var, name, value)
-
-        return cf_name
-
-    def _create_cf_cell_measure_variable(
-        self, cube, dimension_names, cell_measure
-    ):
-        """
-        Create the associated CF-netCDF variable in the netCDF dataset for the
-        given cell_measure.
-
-        Args:
-
-        * cube (:class:`iris.cube.Cube`):
-            The associated cube being saved to CF-netCDF file.
-        * dimension_names (list):
-            Names for each dimension of the cube.
-        * cell_measure (:class:`iris.coords.CellMeasure`):
-            The cell measure to be saved to CF-netCDF file.
-
-        Returns:
-            The string name of the associated CF-netCDF variable saved.
-
-        """
-        # Note: currently shares variable creation code with ancillary-variables.
-        return self._inner_create_cf_cellmeasure_or_ancil_variable(
-            cube, dimension_names, cell_measure
-        )
+    _ELEMENT_TYPE_NAMES = {
+        iris.coords.DimCoord: "coordinate",
+        iris.coords.AuxCoord: "coordinate",
+        iris.coords.CellMeasure: "cell-measure",
+        iris.coords.AncillaryVariable: "ancillary-variable",
+    }
```

Comment:
Good Question.
In subsequent development, I have now got rid of this + just used the class names themselves,
since it's only actually used to generate warning messages inside '_ensure_valid_dtype`.